addons/binding/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/internal/ssl/PinTrustManager.java

```diff
@@ -54,52 +52,33 @@ public void setCallback(PinnedCallback callback) {
     }
 
     @Override
-    public void checkClientTrusted(X509Certificate @Nullable [] chain, @Nullable String authType)
-            throws CertificateException {
-        throw new UnsupportedOperationException();
+    public X509Certificate[] getAcceptedIssuers() {
+        return new X509Certificate[0];
     }
 
-    protected byte[] getEncoded(PinType type, X509Certificate cert) throws CertificateEncodingException {
-        switch (type) {
-            case CERTIFICATE_TYPE:
-                return cert.getEncoded();
-            case PUBLIC_KEY_TYPE:
-                return cert.getPublicKey().getEncoded();
+    @Override
+    public void checkClientTrusted(X509Certificate @Nullable [] certs, @Nullable String authType)
+            throws CertificateException {
+        if ((certs == null) || (certs.length == 0)) {
```

Comment:
Hm, is it really necessary to differentiate between `IllegalArgumentException` and `UnsupportedOperationException`?

Reply:
![grafik](https://user-images.githubusercontent.com/19878123/52504440-c93aed00-2be8-11e9-8c25-4e132b82488f.png)

Reply:
Yes If we would implement this method you are right. But we are throwing `UnsupportedOperationException`. If we'd stick to the API contract we'd need to throw `CertificateException` I guess. 

Your changes are not wrong, I'm not arguing with you. But for better keeping track of project changes it helps if you leave unrelated code parts for a particular problem unchanged. And in this case the operation is really just unsupported, no matter if certificates are presented or not. Just saying, leave this in now if you want to.